core/templates/oa_hash_map.h

```diff
@@ -353,6 +354,13 @@ class OAHashMap {
 		return it;
 	}
 
+	OAHashMap(std::initializer_list<KeyValue<TKey, TValue>> p_init) {
+		reserve(p_init.size());
```

Comment:
I steered you wrong here, OAHashMap has other requirements from its default constructor, and it has to be called here. Just calling reserve() here is not sufficient.